test/unit-tests/testConviva.js

```diff
@@ -0,0 +1,453 @@
+describe('Analytics Framework Conviva Plugin Unit Tests', function() {
+  jest.autoMockOff();
+  //this file is the file that defines TEST_ROOT and SRC_ROOT
+  require("../unit-test-helpers/test_env.js");
+  require("../unit-test-helpers/mock_conviva.js");
+  require(SRC_ROOT + "framework/AnalyticsFramework.js");
+//  require(SRC_ROOT + "plugins/AnalyticsPluginTemplate.js");
+  require(TEST_ROOT + "unit-test-helpers/AnalyticsFrameworkTestUtils.js");
+  require(COMMON_SRC_ROOT + "utils/InitModules/InitOOUnderscore.js");
+
+  var Analytics = OO.Analytics;
+  var Utils = OO.Analytics.Utils;
+  var _ = OO._;
+  var framework;
+
+  var playerName = "Ooyala V4";
+
+  //setup for individual tests
+  var testSetup = function() {
+    framework = new Analytics.Framework();
+    //mute the logging becuase there will be lots of error messages
+    OO.log = function() {
+    };
+  };
+
+  //cleanup for individual tests
+  var testCleanup = function() {
+    OO.Analytics.PluginFactoryList = [];
+    OO.Analytics.FrameworkInstanceList = [];
+    Conviva.currentPlayerStateManager = null;
+    Conviva.currentClient = null;
+    Conviva.currentSystemFactory = null;
+    Conviva.currentContentMetadata = null;
+    //return log back to normal
+//    OO.log = console.log;
+  };
+
+  beforeEach(testSetup);
+  afterEach(testCleanup);
+
+  //helpers
+  var createPlugin = function(framework, metadata) {
+    var convivaPluginFactory = require(SRC_ROOT + "plugins/conviva.js");
+    var plugin = new convivaPluginFactory(framework);
+    plugin.init();
+    metadata = metadata ? metadata : {
+      "gatewayUrl":"testUrl",
+      "customerKey":"testKey"
+    };
+    plugin.setMetadata(metadata);
+    return plugin;
+  };
+
+  it('Test Conviva Plugin Validity', function() {
+    var convivaPluginFactory = require(SRC_ROOT + "plugins/conviva.js");
+    expect(convivaPluginFactory).not.toBeNull();
+    var plugin = new convivaPluginFactory(framework);
+    expect(framework.validatePlugin(plugin)).toBe(true);
```

Comment:
can you also do negative testing with a bad plugin. Make sure that works as well.